tests/ui/crate_level_checks/no_std_swap.rs

```diff
@@ -0,0 +1,23 @@
+#![no_std]
+#![feature(lang_items, start, libc)]
+
+use core::panic::PanicInfo;
+
+#[warn(clippy::all)]
+fn main() {
+    // TODO: do somethjing with swap
+    let mut a = 42;
+    let mut b = 1337;
+
+    a = b;
+    b = a;
+}
+
+#[allow(clippy::empty_loop)]
+#[panic_handler]
+fn panic(_info: &PanicInfo) -> ! {
+    loop {}
+}
+
+#[lang = "eh_personality"]
+extern "C" fn eh_personality() {}
```

Comment:
What is this for?

Reply:
Since we don't explicitly declare unconditional abort on panic, we need to declare `eh_personality` for the `panic_handler` above.

Reply:
Oh, ha TIL. Can we just use `#![crate_type = "lib"]`?

Reply:
Oh, you're right, we can use `#![crate_type = "lib"]` for this. That might be better